src/UseCases/AddComment/AddCommentUseCase.php

```diff
@@ -46,6 +51,10 @@ public function addComment( AddCommentRequest $addCommentRequest ): AddCommentRe
 
 		$donation->addComment( $this->newCommentFromRequest( $addCommentRequest ) );
 
+		if ( !$this->commentTextPassesValidation( $addCommentRequest->getCommentText() ) ) {
```

Comment:
We could store the validation result in a private field and that if the validation was already done. True, the use case object wouldn't be immutable anymore (at least not internally).

I'm just saying, because the validator also checks DNS entries for URLs included in a comment, which would increase the server response time if executed twice. Not really opposing this PR as is, though.